prog/download_boot_image.rb

```diff
@@ -133,13 +133,9 @@ def blob_storage_client
   end
 
   label def start
-    # YYY: we can remove this once we enforce it in the database layer.
-    # Although the default version is used if version is not passed, adding
-    # a sanity check here to make sure version is not passed as nil.
-    fail "Version can not be passed as nil" if version.nil?
```

Comment:
Why are we removing this check? Shouldn't it automatically get the default version if an empty version is passed? So it shouldn't be nil anyway?